### PR TITLE
completed task 4

### PR DESCRIPTION
**Pull request by nehcobiy**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -45,4 +45,51 @@ describe("app", () => {
...
+        .then(({ body }) => {
+          expect(body.length).toBe(13);
+        });
+    });
+    test("array contains objects with correct properties", () => {
+      return request(app)
+        .get("/api/reviews")
+        .expect(200)
+        .then(({ body }) => {
+          body.forEach((review) => {
```
Comment (itskatherine) [inline __tests__/app.test.js]: When you use a forEach, if you hand it an empty array, it skips the function you pass it and the test automatically passes, so passes without actually getting to the expect statements.
For this reason, you should pair the forEach test with an expect statement that checks the returned array has the expected length 🙂

File: __tests__/app.test.js [hunk 2]
```diff
@@ -45,4 +45,51 @@ describe("app", () => {
...
+          });
+        });
+    });
+    test("reviews sorted in descending order of date", () => {
+      return request(app)
+        .get("/api/reviews")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body[0]["created_at"]).toBe("2021-01-25T11:16:54.963Z");
+        });
```
Comment (itskatherine) [inline __tests__/app.test.js:94]: This is just tested the first returned value, the rest could be in the wrong order, I suggest trying out the [jest-sorted](https://www.npmjs.com/package/jest-sorted) library to test if the returned array's in the right order 🙂

File: db/controllers.js [hunk 3]
```diff
@@ -7,3 +7,12 @@ exports.getCategories = (request, response, next) => {
     })
     .catch(next);
 };
+
+exports.getReviews = (request, response, next) => {
+  fetchAllReviews()
+    .then((reviews) => {
+      response.status(200).send(reviews);
```
Comment (itskatherine) [inline db/controllers.js]: Good stuff, but the returned value should be an object with a key of reviews, and a value of the reviews array 🔍

File: db/model.js [hunk 4]
```diff
@@ -5,3 +5,17 @@ exports.fetchAllCategories = () => {
...
 };
+
+exports.fetchAllReviews = () => {
+  const query = `SELECT reviews.owner, reviews.title, reviews.review_id, reviews.category, reviews.review_img_url, reviews.created_at, reviews.votes, reviews.designer,
+  COUNT(comments.review_id) AS comment_count
+  FROM reviews
+  LEFT JOIN comments ON reviews.review_id = comments.review_id
+  GROUP BY reviews.review_id
+  ORDER BY created_at DESC
+  `;
```
Comment (itskatherine) [inline db/model.js:16]: Brilliant! ⭐

**Review by itskatherine — COMMENTED** [review]
Great stuff Yihang - a few minor fixes needed but once you've done them feel free to merge 🙂

File: __tests__/app.test.js (hunk 1 above)
Comment (nehcobiy) [reply __tests__/app.test.js:73]: in the test before this, I have checked for the length of the returned array, is this not sufficient?

Comment (nehcobiy) [reply __tests__/app.test.js]: > When you use a forEach, if you hand it an empty array, it skips the function you pass it and the test automatically passes, so passes without actually getting to the expect statements. For this reason, you should pair the forEach test with an expect statement that checks the returned array has the expected length 🙂

in the test before this, I have checked for the length of the returned array, is this not sufficient?

**Pull request merged**